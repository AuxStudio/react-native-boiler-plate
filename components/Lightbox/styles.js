import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: styleConstants.colors.transBlack,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentContainer: {
    alignSelf: 'center',
    backgroundColor: styleConstants.colors.white,
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    borderRadius: styleConstants.dimensions.borderRadius,
  },
  iconContainer: {
    position: 'absolute',
    top: styleConstants.dimensions.spacing.vertical * 2,
    right: styleConstants.dimensions.spacing.horizontal,
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.white,
  },
});

export default styles;
