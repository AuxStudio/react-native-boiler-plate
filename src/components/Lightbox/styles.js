import { StyleSheet } from 'react-native';

import { colors, dimensions, fonts } from '../../static/styleConstants';

const styles = StyleSheet.create({
  wrapper: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: colors.transBlack,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentContainer: {
    alignSelf: 'center',
    backgroundColor: colors.white,
    paddingVertical: dimensions.spacing.vertical,
    paddingHorizontal: dimensions.spacing.horizontal,
    borderRadius: dimensions.borderRadius,
  },
  iconContainer: {
    position: 'absolute',
    top: dimensions.spacing.vertical * 2,
    right: dimensions.spacing.horizontal,
  },
  icon: {
    fontSize: fonts.sizes.icon,
    color: colors.white,
  },
});

export default styles;
