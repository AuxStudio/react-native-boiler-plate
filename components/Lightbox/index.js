import React from 'react';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import Animator from 'react-native-simple-animators';
import { View, Text, BackHandler } from 'react-native';
import { Touchable } from 'react-native-simple-components';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styleConstants from '../../styleConstants';

import styles from './styles';

import InputContainer from '../InputContainer';

export default class Lightbox extends React.Component {
  constructor(props) {
    super(props);

    this.animateOut = this.animateOut.bind(this);
    this.onBack = this.onBack.bind(this);

    this.state = {
      shouldAnimateOut: false,
    };
  }

  static propTypes = {
    children: PropTypes.node,
    handleClose: PropTypes.func,
    disableClose: PropTypes.bool,
  };

  static defaultProps = {};

  componentDidMount() {
    // Disable back android
    BackHandler.addEventListener('hardwareBackPress', () => {
      return true;
    });
  }

  animateOut() {
    this.setState({
      shouldAnimateOut: true,
    });
  }

  onBack() {
    const { handleClose } = this.props;

    if (handleClose) {
      handleClose();
    } else {
      Actions.pop();
    }
  }

  render() {
    const { shouldAnimateOut } = this.state;
    const { children, disableClose } = this.props;

    const closeComponent = !disableClose ? (
      <Touchable onPress={this.animateOut} style={styles.iconContainer}>
        <Icon name="close" style={styles.icon} />
      </Touchable>
    ) : null;

    return (
      <Animator
        type="translateY"
        initialValue={styleConstants.dimensions.window.height}
        finalValue={0}
        shouldAnimateIn
        shouldAnimateOut={shouldAnimateOut}
        animateOutCallback={this.onBack}
        style={styles.wrapper}
      >
        <InputContainer
          containerStyle={styles.container}
          contentContainerStyle={styles.contentContainer}
        >
          {closeComponent}

          {children}
        </InputContainer>
      </Animator>
    );
  }
}
