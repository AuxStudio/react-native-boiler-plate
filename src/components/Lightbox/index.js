import React from 'react';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import Animator from 'react-native-simple-animators';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { dimensions } from '../../static/styleConstants';

import styles from './styles';

import InputContainer from '../InputContainer';
import Touchable from '../Touchable';

export default class Lightbox extends React.Component {
  constructor(props) {
    super(props);

    this.animateOut = this.animateOut.bind(this);
    this.onBack = this.onBack.bind(this);

    this.state = {
      shouldAnimateOut: false,
    };
  }

  static propTypes = {
    children: PropTypes.node,
    handleClose: PropTypes.func,
    disableClose: PropTypes.bool,
  };

  static defaultProps = {};

  animateOut() {
    this.setState({
      shouldAnimateOut: true,
    });
  }

  onBack() {
    const { handleClose } = this.props;

    if (handleClose) {
      handleClose();
    } else {
      Actions.pop();
    }
  }

  render() {
    const { shouldAnimateOut } = this.state;
    const { children, disableClose } = this.props;

    const closeComponent = !disableClose ? (
      <Touchable
        onPress={this.animateOut}
        style={styles.iconContainer}
        testID="lightbox.button.close"
      >
        <Icon name="close" style={styles.icon} />
      </Touchable>
    ) : null;

    return (
      <Animator
        type="translateY"
        initialValue={dimensions.window.height}
        finalValue={0}
        shouldAnimateIn
        shouldAnimateOut={shouldAnimateOut}
        animateOutCallback={this.onBack}
        style={styles.wrapper}
      >
        <InputContainer
          containerStyle={styles.container}
          contentContainerStyle={styles.contentContainer}
        >
          {closeComponent}

          {children}
        </InputContainer>
      </Animator>
    );
  }
}
