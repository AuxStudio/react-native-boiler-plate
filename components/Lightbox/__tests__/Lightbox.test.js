import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Lightbox from '..';

jest.mock('react-native-simple-animators', () => 'Animator');

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('Lightbox', () => {
  it('renders with all props', () => {
    const component = renderer.create(
      <Lightbox>
        <View />
      </Lightbox>,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    const component = renderer.create(<Lightbox />);

    expect(component).toMatchSnapshot();
  });

  it('should handle animateOut', () => {
    const component = renderer.create(<Lightbox />);
    const instance = component.getInstance();

    instance.animateOut();

    expect(instance.state.shouldAnimateOut).toBe(true);
  });

  it('should handle onBack', () => {
    const component = renderer.create(<Lightbox />);
    const instance = component.getInstance();

    instance.onBack();
  });
});
