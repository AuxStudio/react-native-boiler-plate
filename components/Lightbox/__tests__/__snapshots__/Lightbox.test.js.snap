// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lightbox renders with all props 1`] = `
<Animator
  animateOutCallback={[Function]}
  finalValue={0}
  initialValue={1334}
  shouldAnimateIn={true}
  shouldAnimateOut={false}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  type="translateY"
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "flex": 1,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View />
      </View>
    </RCTScrollView>
  </View>
</Animator>
`;

exports[`Lightbox renders with minimum required props 1`] = `
<Animator
  animateOutCallback={[Function]}
  finalValue={0}
  initialValue={1334}
  shouldAnimateIn={true}
  shouldAnimateOut={false}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  type="translateY"
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "flex": 1,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 15,
              "top": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            Óóç
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</Animator>
`;
